packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -721,24 +729,30 @@ export const RemoteServerFactory = {
                         targetProtocol: 'https'
                     })
 
-                    // We pattern match and add client secrets only to endpoints that
-                    // match the regex specified by options.applySLASPrivateClientToEndpoints
-                    // (see option defaults at the top of this file).
-                    // Other SLAS endpoints, ie. SLAS authenticate (/oauth2/login) and
-                    // SLAS logout (/oauth2/logout), use the Authorization header for a different
-                    // purpose so we don't want to overwrite the header for those calls.
-                    if (incomingRequest.path?.match(options.applySLASPrivateClientToEndpoints)) {
-                        proxyRequest.setHeader('Authorization', `Basic ${encodedSlasCredentials}`)
-                    } else if (!incomingRequest.path?.match(options.slasApiPath)) {
+                    // We don't want the proxy to handle any non-SLAS requests
+                    // or any trusted system requests
+                    if (
+                        !incomingRequest.path?.match(options.slasApiPath) ||
+                        incomingRequest.path?.match(/\/oauth2\/trusted-system/)
+                    ) {
                         const message = `Request to ${incomingRequest.path} is not allowed through the SLAS Private Client Proxy`
                         logger.error(message)
                         return res.status(403).json({
                             message: message
                         })
                     }
 
-                    // /oauth2/trusted-agent/token endpoint requires a different auth header
-                    if (incomingRequest.path?.match(/\/oauth2\/trusted-agent\/token/)) {
+                    // We pattern match and add client secrets only to endpoints that
+                    // match the regex specified by options.applySLASPrivateClientToEndpoints.
+                    //
+                    // Other SLAS endpoints, ie. SLAS authenticate (/oauth2/login) and
+                    // SLAS logout (/oauth2/logout), use the Authorization header for a different
+                    // purpose so we don't want to overwrite the header for those calls.
+                    if (incomingRequest.path?.match(options.applySLASPrivateClientToEndpoints)) {
+                        proxyRequest.setHeader('Authorization', `Basic ${encodedSlasCredentials}`)
```

Comment:
What if I set `options.applySLASPrivateClientToEndpoints` from pwa-kit to have /oauth2/trusted-agent/token in regex form. Would it break?

Reply:
Actually, this is a really good catch. I'll adjust the check so that a very explicit `/trusted-system/token` regex also throws the error